### DISCUSSION TITLE
Limit number of client sessions

### DISCUSSION THREAD

**Discussion by stevie77** [eclipse-milo/milo#1208] [General]
Hi all,
we are using milo to connect to a Inmation OPC UA server and write machine counter data to InfluxDB. We have the problem, that when the OPC server is restarted, the number of sessions explode from 1 to 60.000. Therefore the OPC server gets problems to handle also other requests from other OPC client software (massive performance problems). As soon as we restart our milo client (OPC2Influx), the session is 1 and everything is okay again. Setting the max. size of sessions on the server wasn't successful. So we always have to keep in mind that in case the OPC server was restarted, we also have to restart our client to prevent session explosion.

Does anyone know how to configure milo to use only a maximum amount of sessions? Is it possible by configuration only afterwards the client software was delivered to a customer (not by programming)?
[Inmation Session count for OPC2Influx milo client.pdf](https://github.com/eclipse/milo/files/14008797/Inmation.Session.count.for.OPC2Influx.milo.client.pdf)

**Comment by kevinherron** [comment]
Easiest way to troubleshoot would be to turn the logger for `org.eclipse.milo.opcua.sdk.client.session.SessionFsm` to `DEBUG` level along with getting a Wireshark capture (make sure the connection is not using security).

Then restart the OPC server and let the capture and logs accumulate for a few minutes.

**Comment by stevie77** [comment]
Thanks for your answer. Unfortunately we will not have a network sniffer in a productive environment of our customer. And we cannot reproduce the problem on our local installations (with an other OPC server behind Inmation as Middleware).
Isn't there a possibility to configure Milo (e.g. amount of sessions to be used) in a configuration file?

**Reply by kevinherron** [reply]
There's nothing to configure because it's only one Session per OpcUaClient instance. Any other session accumulation is likely a bug on one side or the other.
